sdk/src/Core/Amazon.Runtime/_netstandard/ClientConfig.cs

```diff
@@ -46,6 +65,34 @@ public IWebProxy GetWebProxy()
             return proxy;
         }
 
+        /// <summary>
+        /// Returns a WebProxy instance to use for HTTPS connections if an
+        /// explicit web proxy hasn't been configured.
+        /// </summary>
```

Comment:
Hey! You're right we probably want to flip these if proxy is set return that, otherwise return `http_proxy` -- I'll go ahead and do that.

---

As for testing I'd totally agree that'd be great, unfortunately I'm not aware of anyway to mock out environment variables in the testing frameworks for only one unit tests. I'd be happy to add them (i'd like more tests in other areas tbf, having to just set the environment variable and running the full integration test suite is unfortunate as your golden pass), but I did search for quite awhile and wasn't able to find a way to do that. Did you have a particular testing strategy in mind?

Reply:
If we would have designed the SDK from the start to allow the environment variables to be mocked it would have been better and that is something we should keep in mind for the future but in the current state we have used this pattern in the tests:

```
var savedEnvVar = Environment.GetEnvironmentVariable("env_var_name");
try
{
   Environment.SetEnvironmentVariable("env_var_name", "some value");
   //Do the test code here

}
finally
{
   Environment.SetEnvironmentVariable("env_var_name", savedEnvVar);
}
```

It saves the value whatever it may be, sets it to the value needed for the test in the try, and then in the finally restores the environment variable to whatever it was before. There are examples of this in this file: https://github.com/aws/aws-sdk-net-staging/blob/6bac6ac7750f63cef6dbf3158b5d32cc9cbde737/sdk/test/UnitTests/Custom/Runtime/ServiceSpecificEndpointTests.cs#L158

Reply:
I've gone ahead and added these tests, and swapped the logic to be explicit first like the comment mentioned. Resolving for now. Feel free to re-open if there's more here!